x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/AuthorizationEngine.java

```diff
@@ -491,6 +493,17 @@ public String toString() {
                 + originatingAuthorizationContext
                 + "]}";
         }
+
+        @Nullable
+        public static String[] indices(TransportRequest transportRequest) {
+            if (transportRequest instanceof IndicesRequest) {
+                final String[] indices = ((IndicesRequest) transportRequest).indices();
+                if (indices != null) {
+                    return ((IndicesRequest) transportRequest).indices();
+                }
+            }
+            return null;
+        }
```

Comment:
I moved this method from the LoggingAuditTrail class to be reused in the assembling of the list of denied indices used in the denied exception.